Sources/ChatManager.swift

```diff
@@ -259,6 +332,62 @@ import NotificationCenter
     }
 }
 
+func reconcileRooms(
+    old: [PCRoom],
+    new: [PCRoom],
+    roomStore: PCRoomStore,
+    delegate: PCChatManagerDelegate
+) {
+    let oldSet = Set(old)
+    let newSet = Set(new)
+
+    let roomsRemovedFrom = oldSet.subtracting(newSet)
+    let roomsAddedTo = newSet.subtracting(oldSet)
+
+    roomsRemovedFrom.forEach { room in
+        let removedRoom = roomStore.removeSync(id: room.id)
```

Comment:
It feels a bit odd that `reconcileRooms` mutates the room store in the removed from room case, but the other cases are all handled elsewhere, and none of the other `reconcileX` functions mutate the stores.

Reply:
Updated and was actually able to massively simplify the user subscription initial state handling logic that was there previously: https://github.com/pusher/chatkit-swift/pull/127/commits/f04ec06b759983cf9c492feb6ec35bef4cefd64f

So thanks!